Replace newlines with spaces in testLevelDescription

CodyReichert opened the pull request:
If the `testLevelDescription` of a `ProcessTransaction` contains
newlines, carriage returns, or line breaks, the generated test suite is
not runnable due to an error like this:

```
it('should respond 200 for "Will send `Authenticated` if authentication is succesful,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

because a normal string in JS can't be multi-line.

This diff runs a replace over the `testLevelDescription` that replaces
these newlines with spaces (" "). A space, as opposed to just stripping
the newlines, keeps things readable in the `it` blocks.

CodyReichert committed:
Replaces newlines with space in testLevelDescription

If the `testLevelDescription` of a `ProcessTransaction` contains
newlines, carriage returns, or line breaks, the generated test suite is
not runnable due to an error like this:

it('should respond 200 for "Will send `Authenticated` if authentication is succesful,
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

because a normal string in JS can't be multi-line.

This diff runs a replace over the `testLevelDescription` that replaces
these newlines with spaces (" "). A space, as opposed to just stripping
the newlines, keeps things readable in the `it` blocks.

CodyReichert commented:
I didn't worry too much about style or moving the `stripNewlines` function into a higher scope, wanted to get your opinion on it first.

But let me know what you think - happy to make any changes.

noahdietz commented on lib/process.js:
```diff
@@ -181,13 +181,18 @@ function processOperations(api, parentPath, topLevel, options) {
     var transactions = []
 
+    // Replace newlines with a space
+    var stripNewlines = function(str) {
```
Let's just put it at the bottom of this file with the other local helpers.

Thanks 😄 !

CodyReichert replied:
Perfect, moved the function to the bottom (and renamed it to `replaceNewlines` to be more indicative of the actual operation).

noahdietz reviewed (changes requested):
Looks good, see comment for requested change. 

Also, please add the `jsdoc` comments to the helper with a description that includes how this helper is used WRT processing the spec, i.e. `used to replace newlines with a space in each response.description` or something like that. 

Commit this change, then as a separate commit run `npm run gen-docs` and commit the `docs/` changes! (I need to add this step to the contib...)

CodyReichert committed:
Cleanup replaceNewlines helper in process module

Update changes in lib/process per review:

- Move replaceNewlines to it's own helper function
- Add jsdoc comments to replaceNewlines function

CodyReichert committed:
Regenerate docs for new function

Update docs to add the new 'replaceNewlines' function in the lib/process
module.

CodyReichert commented:
@noahdietz - thanks, I added and updated the docs for the new function. I'm not super familiar w/ jsdoc, so just let me know if I missed anything.

noahdietz commented:
Thanks for the changes @CodyReichert 

Merging and I will release a new patch revision of this project.
